linked-list/src/example/java/Element.java

```diff
@@ -0,0 +1,52 @@
+public final class Element {
+    private final Object value;
+    private Element prev;
+    private Element next;
+
+    public Element(Object value, Element prev, Element next) {
+        this.value = value;
+        this.prev = prev;
+        this.next = next;
+    }
+
+    public Element getPrev() {
+        return prev;
+    }
+
+    public void setPrev(Element prev) {
+        this.prev = prev;
+    }
+
+    public Element getNext() {
+        return next;
+    }
+
+    public void setNext(Element next) {
+        this.next = next;
+    }
+
+    public Object getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Since the java implementation (heavily inspired by the ruby implementation) creates a circular list, we need the next-setters during `shift()`. The value-getter is for shifting and popping.

Though, you're right, we don't need the `equals()` or `hashCode()` implementations. We compare pointers when checking for circular references not the values.